File: src/main/java/rocks/sakira/gentlefawn/register/Entities.java
package rocks.sakira.gentlefawn.register;

import net.minecraft.entity.EntityClassification;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.ai.attributes.Attributes;
import net.minecraft.entity.ai.attributes.GlobalEntityTypeAttributes;
import net.minecraft.entity.passive.AnimalEntity;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import rocks.sakira.gentlefawn.GentleFawn;
import rocks.sakira.gentlefawn.entity.EntityDeer;

public class Entities {
    public static final DeferredRegister<EntityType<?>> REGISTER = DeferredRegister.create(ForgeRegistries.ENTITIES, GentleFawn.MOD_ID);

    public static final RegistryObject<EntityType<EntityDeer>> DEER_ENTITY = REGISTER.register(
            "deer",

            () -> EntityType.Builder
                    .create(EntityDeer::new, EntityClassification.CREATURE)
                    .size(0.9F, 1.9F)
                    .setTrackingRange(64)
                    .setUpdateInterval(1)
                    .build("deer")
    );

    @SubscribeEvent
    public static void registerEntities(final FMLCommonSetupEvent event) {
        GlobalEntityTypeAttributes.put(
                DEER_ENTITY.get(),

                AnimalEntity.func_233666_p_()
                        .func_233815_a_(Attributes.field_233818_a_, 8.0D)  // Max health
                        .func_233815_a_(Attributes.field_233821_d_, 0.16D)  // Movement speed
                        .func_233813_a_()
        );
    }
}

File: src/main/kotlin/io/github/faecraft/gentlefawn/register/Entities.kt
package io.github.faecraft.gentlefawn.register

import io.github.faecraft.gentlefawn.Common.MOD_ID
import io.github.faecraft.gentlefawn.entity.DeerEntity
import net.fabricmc.fabric.api.`object`.builder.v1.entity.FabricDefaultAttributeRegistry
import net.fabricmc.fabric.api.`object`.builder.v1.entity.FabricEntityTypeBuilder
import net.minecraft.entity.EntityDimensions
import net.minecraft.entity.SpawnGroup
import net.minecraft.entity.attribute.EntityAttributes
import net.minecraft.entity.passive.AnimalEntity
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object Entities {
    val DEER = Registry.register(
        Registry.ENTITY_TYPE,
        Identifier(MOD_ID, "deer"),

        FabricEntityTypeBuilder
            .create(SpawnGroup.AMBIENT, ::DeerEntity)
            .dimensions(EntityDimensions.changing(0.9F, 1.9F))
            .trackRangeBlocks(64)
            .trackedUpdateRate(1)
            .build()
    )

    fun registerAttributes() {
        FabricDefaultAttributeRegistry.register(
            DEER,
            AnimalEntity.createMobAttributes()
                .add(EntityAttributes.GENERIC_MAX_HEALTH, 8.0)
                .add(EntityAttributes.GENERIC_MOVEMENT_SPEED, 0.16)
        )
    }
}
